src/main/java/io/openliberty/tools/common/plugins/util/DevUtil.java

```diff
@@ -5701,13 +5701,22 @@ public void writeDevcMetadata(boolean alive) {
             metadataWriter.writeStartElement("devcModeMetaData");
             writeElement(metadataWriter, "containerEngine", isDocker ? DEVC_CONTAINER_DOCKER : DEVC_CONTAINER_PODMAN);
             writeElement(metadataWriter, "containerName", containerName != null ? containerName : DEVMODE_CONTAINER_BASE_NAME);
+            writeElement(metadataWriter, "imageName", imageName);
+            if (containerfile != null) {
+                writeElement(metadataWriter, "containerfile", containerfile.getCanonicalPath());
+            }
+            if (containerBuildContext != null) {
+                writeElement(metadataWriter, "containerBuildContext", containerBuildContext.getCanonicalPath());
+            }
             writeElement(metadataWriter, "containerAlive", String.valueOf(alive));
+            writeElement(metadataWriter, "containerBuildTimeout", Integer.toString(containerBuildTimeout));
+            writeElement(metadataWriter, "containerRunOpts", containerRunOpts);
```

Comment:
Are we sure `containerRunOpts` will not be null? What about from LGP?

Reply:
We skip writing the value for the elements if the property is null. The element will be in the file, just empty.

Reply:
The write and null check are here: https://github.com/OpenLiberty/ci.common/blob/e69e5a37508715a3ba91fe504398d5e8f560dcc0/src/main/java/io/openliberty/tools/common/plugins/util/DevUtil.java#L5723-L5727